### PR TITLE
Added BlockCacheManagerInterface and its implementation

### PR DESCRIPTION
**Pull request by spolischook**
Related to this issue https://github.com/sonata-project/SonataBlockBundle/issues/159

### CODE REVIEW

File: Cache/BlockCacheManager.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+namespace Sonata\BlockBundle\Cache;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\BlockBundle\Block\BlockContextManagerInterface;
+use Sonata\Cache\CacheManagerInterface;
+use Doctrine\Common\Util\ClassUtils;
+
+class BlockCacheManager implements BlockCacheManagerInterface
+{
+    /** @var  array */
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Should looks like:

```php
/**
 * @var array
 */
```

File: Cache/BlockCacheManager.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    protected $cacheBlocks;
+
+    /** @var  BlockContextManagerInterface */
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Should looks like:

```php
/**
 * @var array
 */
```

File: Cache/BlockCacheManager.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    protected $blockContextManager;
+
+    /** @var  CacheManagerInterface */
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Should looks like:

```php
/**
 * @var array
 */
```

File: Cache/BlockCacheManager.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setBlockToCache(BlockInterface $block)
+    {
+        // TODO: Implement setBlockToCache() method.
+    }
+
+    /** {@inheritdoc} */
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Should looks like:

```php
/**
 * {@inheritdoc}
 */
```

File: Cache/BlockCacheManager.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        $cacheKeys = array_merge(
+            $cacheKeys,
+            $blockContext->getSetting('extra_cache_keys')
+        );
+        $cacheElement = $cacheService->get($cacheKeys);
+
+        return $cacheElement->getData() ? $cacheElement->getData() : false;
+    }
+
+    /** {@inheritdoc} */
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Should looks like:

```php
/**
 * {@inheritdoc}
 */
```

File: Cache/BlockCacheManager.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
+<?php
+
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Missing Sonata headers.

File: Cache/BlockCacheManagerInterface.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<?php
+
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManagerInterface.php]: Missing Sonata headers.

File: Cache/BlockCacheManager.php [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+namespace Sonata\BlockBundle\Cache;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\BlockBundle\Block\BlockContextManagerInterface;
+use Sonata\Cache\CacheManagerInterface;
+use Doctrine\Common\Util\ClassUtils;
+
+class BlockCacheManager implements BlockCacheManagerInterface
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: Missing PHP doc block.

File: Cache/BlockCacheManagerInterface.php [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+namespace Sonata\BlockBundle\Cache;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+
+interface BlockCacheManagerInterface
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManagerInterface.php]: Missing PHP doc block.

**Comment by Soullivaneuh** [comment]
@spolischook could you please rebase your PR to get new Travis config on it?

**Comment by Soullivaneuh** [comment]
Please run `make cs` to fix CS issues: https://travis-ci.org/sonata-project/SonataBlockBundle/jobs/66814609#L210

**Comment by spolischook** [comment]
@Soullivaneuh  done

**Comment by Soullivaneuh** [comment]
Sorry, seems not: https://travis-ci.org/sonata-project/SonataBlockBundle/jobs/66837820#L209

Maybe wrong version of php-cs-fixer. Could you try `composer update` before `make cs`?

**Comment by Soullivaneuh** [comment]
Tests passes.

I let @rande validate this PR.

**Comment by OskarStark** [comment]
@rande anything to do on this PR?

File: Cache/BlockCacheManager.php [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+namespace Sonata\BlockBundle\Cache;
+
+use Doctrine\Common\Util\ClassUtils;
+use Sonata\BlockBundle\Block\BlockContextManagerInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\Cache\CacheManagerInterface;
+
+/**
+ * Class BlockCacheManager.
```
Comment (Soullivaneuh) [inline Cache/BlockCacheManager.php]: This comment is useless. We already now this is the BlockCacheManager class. :-)

Just set your `@author` tag if you have nothing to tell more. :+1:

File: Cache/BlockCacheManager.php [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        if (!$cacheServiceId) {
+            return false;
+        }
+
+        return $this->cacheManager->getCacheService($cacheServiceId);
+    }
+
+    public function setCacheManager(CacheManagerInterface $cacheManager)
```
Comment (core23) [inline Cache/BlockCacheManager.php]: phpdoc missing

Comment (spolischook) [inline Cache/BlockCacheManager.php]: fixed

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**